app/lib/service/download_counts/backend.dart

```diff
@@ -19,7 +27,59 @@ DownloadCountsBackend get downloadCountsBackend =>
 class DownloadCountsBackend {
   final DatastoreDB _db;
 
-  DownloadCountsBackend(this._db);
+  late CachedValue<Map<String, int>> _thirtyDaysTotals;
+  var _lastData = (data: <String, int>{}, etag: '');
+
+  DownloadCountsBackend(this._db) {
+    _thirtyDaysTotals = CachedValue(
+        name: 'thirtyDaysTotalDownloadCounts',
+        maxAge: Duration(days: 14),
+        interval: Duration(minutes: 30),
+        updateFn: _updateThirtyDaysTotals);
+  }
+
+  Future<Map<String, int>> _updateThirtyDaysTotals() async {
+    try {
+      final info = await storageService
+          .bucket(activeConfiguration.reportsBucketName!)
+          .info(downloadCounts30DaysTotalsFileName);
```

Comment:
IIRC we have a tryInfo in the `lib/shared/storage.dart` which could at least handle the 404 part of the exceptions.